lib/KTable/index.vue

```diff
@@ -0,0 +1,484 @@
+                    // Allow default behavior when reaching the first cell
+                    return;
+                  }
+                }
+              }
+            } else {
+              if (!event.shiftKey) {
+                if (colIndex < totalCols - 1) {
+                  nextColIndex = colIndex + 1;
+                } else if (rowIndex < totalRows - 1) {
+                  nextColIndex = 0;
+                  nextRowIndex = rowIndex + 1;
+                } else {
+                  // Allow default behavior when reaching the last cell
+                  return;
+                }
+              } else {
+                if (colIndex > 0) {
+                  nextColIndex = colIndex - 1;
+                } else if (rowIndex > 0) {
+                  nextColIndex = totalCols - 1;
+                  nextRowIndex = rowIndex - 1;
+                } else {
+                  // Allow default behavior when reaching the first cell
+                  return;
+                }
+              }
+            }
+
+            break;
+
+          default:
+            return;
+        }
+
+        this.focusCell(nextRowIndex, nextColIndex);
+        this.focusedRowIndex = nextRowIndex === -1 ? null : nextRowIndex;
+        this.focusedColIndex = nextColIndex;
+
+        this.highlightHeader(nextColIndex);
+
+        event.preventDefault();
+      },
+      getCell(rowIndex, colIndex) {
+        if (rowIndex === -1) {
+          return this.$el.querySelector(`thead th:nth-child(${colIndex + 1})`);
+        } else {
+          return this.$el.querySelector(
+            `tbody tr:nth-child(${rowIndex + 1}) td:nth-child(${colIndex + 1})`
+          );
+        }
+      },
+      scrollCellIntoView(cell) {
+        if (cell) {
+          cell.scrollIntoView({ behavior: 'smooth', block: 'nearest', inline: 'nearest' });
+
+          // Adjust scroll position to account for sticky headers
+          const stickyHeader = this.$el.querySelector('.sticky-header');
+          const stickyColumn = this.$el.querySelector('.sticky-column');
+          const tableWrapper = this.$refs.tableWrapper;
+
+          const stickyHeaderHeight = stickyHeader ? stickyHeader.offsetHeight : 0;
+          const stickyColumnWidth = stickyColumn ? stickyColumn.offsetWidth : 0;
+
+          const cellRect = cell.getBoundingClientRect();
+          const wrapperRect = tableWrapper.getBoundingClientRect();
+
+          if (cellRect.top < wrapperRect.top + stickyHeaderHeight) {
+            tableWrapper.scrollTop -= wrapperRect.top + stickyHeaderHeight - cellRect.top;
+          }
+          if (cellRect.left < wrapperRect.left + stickyColumnWidth) {
+            tableWrapper.scrollLeft -= wrapperRect.left + stickyColumnWidth - cellRect.left;
+          }
+        }
+      },
+      focusCell(rowIndex, colIndex) {
+        let nextCell;
+        if (rowIndex === -1) {
+          nextCell = this.$el.querySelector(`thead th:nth-child(${colIndex + 1})`);
```

Comment:
I think its better to have vue refs to the elements to make this operation more lighter to the browser, as this method is going to be called on every keypress.

Reply:
done